### PR TITLE
py-scikit-image: add new version

### PR DESCRIPTION
**Pull request by archxlith**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 1]
```diff
@@ -13,13 +13,15 @@ class PyScikitImage(PythonPackage):
...
     pypi = "scikit-image/scikit-image-0.17.2.tar.gz"
 
+    version('0.18.1', sha256='fbb618ca911867bce45574c1639618cdfb5d94e207432b19bc19563d80d2f171')
     version('0.17.2', sha256='bd954c0588f0f7e81d9763dc95e06950e68247d540476e06cb77bcbcd8c2d8b3')
     version('0.14.2', sha256='1afd0b84eefd77afd1071c5c1c402553d67be2d7db8950b32d6f773f25850c1f')
     version('0.12.3', sha256='82da192f0e524701e89c5379c79200bc6dc21373f48bf7778a864c583897d7c7')
 
     extends('python', ignore=r'bin/.*\.py$')
 
     depends_on('python@3.6:', when='@0.16.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py]:
```suggestion
    depends_on('python@3.7:', when='@0.18:', type=('build', 'run'))
    depends_on('python@3.6:', when='@0.16.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 2]
```diff
@@ -13,13 +13,15 @@ class PyScikitImage(PythonPackage):
...
+    depends_on('py-setuptools', type='build')
     depends_on('py-cython@0.23.4:', type='build')
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('py-numpy@1.14.1:', type=('build', 'run'), when='@0.16.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py]: Can you add:
```python
depends_on('py-numpy@1.16.5:', type=('build', 'run'), when='@0.18:')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by archxlith** [comment]
Made requested change.

File: var/spack/repos/builtin/packages/py-scikit-image/package.py (hunk 1 above)
Comment (archxlith) [reply var/spack/repos/builtin/packages/py-scikit-image/package.py]: python version dependencies updated.

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 3]
```diff
@@ -13,17 +13,21 @@ class PyScikitImage(PythonPackage):
...
 
+    version('0.18.1', sha256='fbb618ca911867bce45574c1639618cdfb5d94e207432b19bc19563d80d2f171')
     version('0.17.2', sha256='bd954c0588f0f7e81d9763dc95e06950e68247d540476e06cb77bcbcd8c2d8b3')
     version('0.14.2', sha256='1afd0b84eefd77afd1071c5c1c402553d67be2d7db8950b32d6f773f25850c1f')
     version('0.12.3', sha256='82da192f0e524701e89c5379c79200bc6dc21373f48bf7778a864c583897d7c7')
 
     extends('python', ignore=r'bin/.*\.py$')
 
-    depends_on('python@3.6:', when='@0.16.1:')
+    depends_on('python@3.7:', when='@0.18:', type=('build', 'run')))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py]:
```suggestion
    depends_on('python@3.7:', when='@0.18:', type=('build', 'run'))
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-scikit-image/package.py]: Ping @archxlith

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 4]
```diff
@@ -13,17 +13,21 @@ class PyScikitImage(PythonPackage):
...
+    version('0.18.1', sha256='fbb618ca911867bce45574c1639618cdfb5d94e207432b19bc19563d80d2f171')
     version('0.17.2', sha256='bd954c0588f0f7e81d9763dc95e06950e68247d540476e06cb77bcbcd8c2d8b3')
     version('0.14.2', sha256='1afd0b84eefd77afd1071c5c1c402553d67be2d7db8950b32d6f773f25850c1f')
     version('0.12.3', sha256='82da192f0e524701e89c5379c79200bc6dc21373f48bf7778a864c583897d7c7')
 
     extends('python', ignore=r'bin/.*\.py$')
 
-    depends_on('python@3.6:', when='@0.16.1:')
+    depends_on('python@3.7:', when='@0.18:', type=('build', 'run'))
+    depends_on('python@3.6:', when='@0.16.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py]:
```suggestion
    depends_on('python@3.7:', when='@0.18:', type=('build', 'link', 'run'))
    depends_on('python@3.6:', when='@0.16.1:', type=('build', 'link', 'run'))
```

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 5]
```diff
@@ -13,17 +13,21 @@ class PyScikitImage(PythonPackage):
...
 
-    depends_on('python@3.6:', when='@0.16.1:')
+    depends_on('python@3.7:', when='@0.18:', type=('build', 'link', 'run'))
+    depends_on('python@3.6:', when='@0.16.1:', type=('build', 'link', 'run'))
+    depends_on('py-setuptools', type='build')
     depends_on('py-cython@0.23.4:', type='build')
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('py-numpy@1.14.1:', type=('build', 'run'), when='@0.16.1:')
     depends_on('py-numpy@1.15.1:', type=('build', 'run'), when='@0.17.1:')
+    depends_on('py-numpy@1.16.5:', type=('build', 'run'), when='@0.18:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py]:
```suggestion
    depends_on('py-numpy@1.14.1:', type=('build', 'link', 'run'), when='@0.16.1:')
    depends_on('py-numpy@1.15.1:', type=('build', 'link', 'run'), when='@0.17.1:')
    depends_on('py-numpy@1.16.5:', type=('build', 'link', 'run'), when='@0.18:')
```

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Successfully builds on macOS 10.15.7 with Python 3.8.8 and Apple Clang 12.0.0. The previous version of `py-scikit-image` seems to have some build errors, not sure why.

**Pull request merged**